util.go

```diff
@@ -64,6 +68,42 @@ func init() {
 		AwaitingPulseJobUpdateStates[status] = true
 	}
 }
+func createCertPool(path string, extensions map[string]struct{}) (*x509.CertPool, error) {
+	certPool := x509.NewCertPool()
+
+	_, err := os.Stat(path)
+	if err != nil {
+		return nil, errors.New("given certs path doesn't exist")
+	}
+
+	caFiles, err := ioutil.ReadDir(path)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(caFiles) == 0 {
+		return nil, errors.New("no possible certs found in " + path)
+	}
+
+	for _, cert := range caFiles {
+		// Skip directories
+		if cert.IsDir() {
+			continue
+		}
+
+		// Skip any files that do not contain the right extension
+		if _, ok := extensions[filepath.Ext(cert.Name())]; !ok {
+			continue
+		}
+
+		caCert, err := ioutil.ReadFile(filepath.Join(path, cert.Name()))
+		if err != nil {
+			return nil, err
+		}
+		certPool.AppendCertsFromPEM(caCert)
```

Comment:
This function returns a boolean to indicate if the operation was successful -- https://golang.org/pkg/crypto/x509/#CertPool.AppendCertsFromPEM

It should be either checked or explicitly discarded.

Reply:
Using it to detect cases where no certificates get loaded into the pool and returning an error based upon that.